internal/indexer/protocol.go

```diff
@@ -30,39 +30,40 @@ func rangeForObject(obj types.Object, pos token.Position) (protocol.Pos, protoco
 	return start, end
 }
 
-// toMarkedString creates a protocol.MarkedString object from the given content. The signature
+// toMarkupContent creates a protocol.MarkupContent object from the given content. The signature
 // and extra parameters are formatted as code, if supplied. The docstring is formatted as markdown,
 // if supplied.
-func toMarkedString(signature, docstring, extra string) (mss []protocol.MarkedString) {
-	for _, m := range []*protocol.MarkedString{formatCode(signature), formatMarkdown(docstring), formatCode(extra)} {
-		if m != nil {
-			mss = append(mss, *m)
+func toMarkupContent(signature, docstring, extra string) (mss protocol.MarkupContent) {
+	var ss []string
+
+	for _, m := range []string{formatCode(signature), formatMarkdown(docstring), formatCode(extra)} {
+		if m != "" {
+			ss = append(ss, m)
 		}
 	}
 
-	return mss
+	return protocol.NewMarkupContent(strings.Join(ss, "\n\n---\n\n"), protocol.Markdown)
 }
 
-// formatMarkdown creates a protocol.MarkedString object containing a markdown-formatted version
+// formatMarkdown creates a string containing a markdown-formatted version
 // of the given string. If the given string is empty, nil is returned.
```

Comment:
just as a nit, here and L61 weren't updated (by me :hand:) to reflect the change in what gets returned